src/main/java/ca/spottedleaf/starlight/mixin/common/lightengine/ThreadedLevelLightEngineMixin.java

```diff
@@ -97,9 +97,7 @@ private void queueTaskForSection(final int chunkX, final int chunkY, final int c
             for (int dz = -1; dz <= 1; ++dz) {
                 ChunkHolder neighbour = world.getChunkSource().chunkMap.getUpdatingChunkIfPresent(CoordinateUtils.getChunkKey(dx + chunkX, dz + chunkZ));
                 if (neighbour != null) {
-                    neighbour.chunkToSave = neighbour.chunkToSave.thenCombine(updateFuture, (final ChunkAccess curr, final Void ignore) -> {
-                        return curr;
-                    });
+                    neighbour.chunkToSave = updateFuture.thenCompose(v -> neighbour.chunkToSave);
```

Comment:
Looking over the code now, you could actually just remove this logic - the ticket should keep the chunks from unloading.

Reply:
Alright, I'll remove the logic ASAP :)

Still, I wonder what the original rationale for this code was — under what conditions does the server consider a world safe to save? Considering this code applies to neighbours, would it be necessary to add tickets for the neighbours as well?

Reply:
The chunk cannot unload until the chunkToSave is completed, so the idea was to stop the unload until the light task was done. But the ticket level should ensure that the chunk shouldn't unload, and the ticket cannot be removed until at least the light task is done.
Since the ticket level is 33, the neighbours will have ticket level 34. Chunks only unload once they exceed somewhere around 46, I don't remember off the top of my head, but given the chunk and its neighbours are above the threshold they wont be unloaded.

Reply:
Ah, thanks for the explanation!

Reply:
This should be resolved with 334500ca6e9c522924104586cac2e563430ec7ce :)